lib/rules/no-duplicate-id.js

```diff
@@ -1,16 +1,67 @@
 'use strict';
 
+const AstNodeInfo = require('../helpers/ast-node-info');
 const NodeMatcher = require('../helpers/node-matcher');
 const Rule = require('./base');
 
+class ConditionalScope {
+  constructor() {
+    this.frames = [new Set()];
+
+    this.conditionals = [];
+  }
+
+  pushConditional() {
+    this.conditionals.push(new Set());
+  }
+
+  popConditional() {
+    let idsWithinConditional = this.conditionals.pop();
+
+    if (this.conditionals.length > 0) {
+      let parentConditional = this.conditionals[this.conditionals.length - 1];
+      idsWithinConditional.forEach((idValue) => {
```

Comment:
Is `idValue` actually `elementId`? The name feels a bit confusing to me, since it required me to ask you what it represented.